lib/fluent/plugin/in_tcp.rb

```diff
@@ -83,7 +85,32 @@ def start
               router.emit(tag, time, record)
             end
           end
-          conn.buffer.slice!(0, pos) if pos > 0
+          buf.slice!(0, pos) if pos > 0
+        end
+      else
+        server_create(:in_tcp_server_batch_emit, @port, bind: @bind, resolve_name: !!@source_hostname_key) do |data, conn|
+          conn.buffer << data
+          buf = conn.buffer
+          pos = 0
+          es = Fluent::MultiEventStream.new
+          while i = buf.index(@delimiter, pos)
+            msg = buf[pos...i]
+            pos = i + del_size
+
+            @parser.parse(msg) do |time, record|
+              unless time && record
+                log.warn "pattern not matched", message: msg
+                next
+              end
+
+              time ||= extract_time_from_record(record) || Fluent::EventTime.now
+              record[@source_address_key] = conn.remote_addr if @source_address_key
+              record[@source_hostname_key] = conn.remote_host if @source_hostname_key
+              es.add(time, record)
+            end
+          end
+          router.emit_stream(@tag, es)
```

Comment:
This PR has already been merged. but I found a considering point.

if `es` becomes bigger than [Buffer#chunk_size_over?](https://github.com/fluent/fluentd/blob/2ad5b6de80cdd64d2ba27526f806e91323e12a68/lib/fluent/plugin/buffer.rb#L564), fluentd shows warning log and does rollback constantly.
so I think it needs to add a configuration parameter to set the size of stream or something.

Reply:
> so I think it needs to add a configuration parameter to set the size of streams or something.

re-think carefully, In this situation,  we can set`chunk_limit_size`. so it's not problem.